### ISSUE TITLE
Add a system test to check acceptance of agents with higher version

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh-qa#4371]
| Target version | Related issue | Related PR/dev branch |
|--------------------|--------------------|-----------------|
|                        |https://github.com/wazuh/wazuh/issues/18138|                    |

## Description

The issue above asks for the agent version accepting filter to be optional. We need to check this feature.

## Configurations

The configuration is shown at https://github.com/wazuh/wazuh/issues/18138.

## Proposed test cases

Let this case list be a suggestion:

### With `<allow_higher_versions>` set to `yes`:

- [ ] The manager shall **accept** agents with _version< manager_ (4.5.2).
- [ ] The manager shall **accept** agents with _version = manager_ (4.6.0).
- [ ] The manager shall **accept** agents with _minor = manager_ and _patch > manager_ (4.6.1).
- [ ] The manager shall **accept** agents with _minor > manager_ (4.7.0).

### With `<allow_higher_versions>` set to `no`:

- [ ] The manager shall **accept** agents with _version < manager_ (4.5.2).
- [ ] The manager shall **accept** agents with _version = manager_ (4.6.0).
- [ ] The manager shall **accept** agents with _minor = manager_ and _patch > manager_ (4.6.1).
- [ ] The manager shall **reject** agents with _minor > manager_ (4.7.0).

## Considerations

In conclusion:

- With `<allow_higher_versions>` set to `yes`, the filter is disabled. Thus, all versions must be accepted.
- With `<allow_higher_versions>` set to `no`:
  - Older versions of agents are always accepted.
  - The same _major.minor_ version is always allowed, no matter the patch version.
- The acceptance of higher major versions is unchecked (unknown).

**Comment by davidjiglesias** [comment]
Maybe this could be part of some future upgrades for our Deployability tests.
